validator/src/network/rest/mod.rs

```diff
@@ -1,29 +1,59 @@
+use crate::services::mixmining;
 use iron::prelude::*;
+use presence::mixnode;
+use presence::topology;
 use router::Router;
+use std::sync::{Arc, Mutex};
```

Comment:
I might be totally wrong about this, however, based on some comments I've read on the internet (so the most reliable source on the planet), you want to be using this mutex: https://docs.rs/futures/0.3.4/futures/lock/struct.Mutex.html . As from what I understand, `std::sync::Mutex` would block the whole thread what could lead to blocking async executor hence causing a deadlock. Meanwhile `futures::lock::Mutex` would simply return `Poll::Pending` and cause the Future (or tokio task) to be polled again at later time.